### PR TITLE
DOCTOKITTY 🐙🐈🙀

### PR DESCRIPTION
**Pull request by emplums**
This PR refactors our documentation site to use the brand new ✨ [doctocat](https://github.com/primer/doctocat) ✨ 

I've realized that ideally we'd need to use Yarn Workspaces here so that we can symlink between the components used in the LiveEditors and components in the `src` folder of the package. Otherwise, we're stuck with having the published version of the components in the LiveEditors which means we can't use the docs site as part of our development process.

I started migrating to Yarn Workspaces, but it's a ton of file changes, so I think it would be best to have this PR reviewed and merged into a release branch, then do a separate PR for the Yarn Workspaces branch.

### CODE REVIEW

File: docs/components/constants.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import * as styledSystem from 'styled-system'
```
Comment (emplums) [inline docs/components/constants.js]: Having this file here is temporary until we setup yarn workspaces.

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# Doctocat Template
```
Comment (colebemis) [inline docs/README.md]: We can probably delete this file.

File: docs/components/constants.js (hunk 1 above)
Comment (colebemis) [reply docs/components/constants.js]: ^ We should probably write a code comment that mentions that this file is temporary.

File: docs/content/components/Avatar.md [hunk 3]
```diff
@@ -31,4 +33,4 @@ Avatar components get `COMMON` system props. Read our [System Props](/components
 
 export const meta = {
   displayName: 'Avatar'
```
Comment (colebemis) [inline docs/content/components/Avatar.md]: Now that we're using Doctocat, we don't need the `meta` export anymore.

File: docs/content/components/Tooltip.md [hunk 4]
```diff
@@ -1,8 +1,11 @@
-# Tooltip
+---
+title: Tooltip
+---
 
 The Tooltip component adds a tooltip to add context to elements on the page. The Tooltip has a black background by default.
 
-### ⚠️ Usage warning! ⚠️
+***⚠️ Usage warning! ⚠️***
```
Comment (colebemis) [inline docs/content/components/Tooltip.md]: This doesn't matter but did you mean to make this bold and italic?

File: now.json [hunk 5]
```diff
@@ -1,23 +1,21 @@
...
     "rollup.config.js",
     "src",
     "static"
+  ],
+  "builds": [
+    {
+      "src": "docs/package.json",
+      "use": "@now/static-build",
+      "config": {"distDir": "public"}
+    }
```
Comment (colebemis) [inline now.json]: We should set up a `/components` path prefix so the `primer.style/components` domain continues to work. Here's the documentation for that: https://primer.style/doctocat/usage/deployment#adding-a-primerstyle-path-alias

File: docs/package.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "doctocat-template",
```
Comment (colebemis) [inline docs/package.json]:
```suggestion
  "name": "docs",
```

File: docs/package.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  "repository": "primer/doctocat-template",
```
Comment (colebemis) [inline docs/package.json]: We need to update the repository field to make the edit links work. 
```suggestion
  "repository": "primer/components",
```

File: docs/package.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  "version": "1.0.0",
+  "scripts": {
+    "clean": "gatsby clean",
+    "develop": "gatsby develop",
+    "build": "npm i && gatsby build"
```
Comment (colebemis) [inline docs/package.json]: Is the `npm i` necessary to include in the build script?

File: docs/package.json [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  },
```
Comment (colebemis) [inline docs/package.json]: ~~Do we need to define a `now-build` script?~~
Oh, just looked at the docs and we don't need to define a `now-build` script to make the now builder work  🎉Very cool

**Comment by emplums** [comment]
@colebemis I believe this is ready for one last review + poke around! There are a few things that I'm leaving as to do's:

- [ ] Get cool animation back in the Hero. This was done with some next libraries and I didn't have time to refactor in this go around
- [ ] I had a hard time formatting the theme in the theme docs, I'm not sure if we really need it there though 🤔 For now I've just linked to the file on GitHub

**Comment by emplums** [comment]
Note: `focus-lock` doesn't love the shadow dom stuff being done in @reach/dialog and throws an error when the dialog is opened https://github.com/theKashey/focus-lock/issues/11

**Comment by emplums** [comment]
Known issues:

- Details example gets cut off in the the LiveEditor
- Sticky example doesn't work

These are both due to https://github.com/primer/doctocat/issues/89 which hopefully we can address soon!

File: docs/content/BorderBox.md [hunk 10]
```diff
@@ -1,16 +1,19 @@
...
+title: BorderBox
+---
+
 
 BorderBox is a Box component with a border. When no `borderColor` is present, the component defaults to a gray border.
 
 ## Default example
 
-```.jsx
+```jsx live live
```
Comment (dmarcey) [inline docs/content/BorderBox.md]: Do these need to be "live live"? Maybe a find/replace typo?
Doesn't look like it's hurting anything though.

**Comment by dmarcey** [comment]
I played around with the deployed site a bit and it LGTM!
🚀 🚀 🚀

**Comment by emplums** [comment]
Addressed the above issues!

LET'S DO THIS (FINALLY!!!) 🎉

**Pull request merged**